In [2]:
from google.colab import files
uploaded = files.upload()

Saving Day11_Messy_Company_Employee_Dataset.csv to Day11_Messy_Company_Employee_Dataset.csv


In [3]:
import pandas as pd

# 1. Load the dataset
df = pd.read_csv("Day11_Messy_Company_Employee_Dataset.csv")

# 2. Inspect initial structure and missing values
print("--- Initial Dataset Shape ---")
print(df.shape)
print("\n--- Missing Values Before Cleaning ---")
print(df.isnull().sum())

# 3. Handle Missing Values
# Impute numerical columns using their median values
num_cols = ['Age', 'Annual_Salary', 'Experience_Years', 'Performance_Score']
for col in num_cols:
    df[col] = df[col].fillna(df[col].median())

# Impute categorical columns using their mode (most frequent value)
cat_cols = ['Department', 'Gender', 'City', 'Work_Mode']
for col in cat_cols:
    df[col] = df[col].fillna(df[col].mode()[0])

# 4. Identify and Remove Duplicates
print(f"\nTotal duplicate rows detected: {df.duplicated().sum()}")
df = df.drop_duplicates()

# 5. Final Verification
print("\n--- Missing Values After Cleaning ---")
print(df.isnull().sum())
print("\n--- Final Cleaned Shape ---")
print(df.shape)

# 6. Export the Cleaned Dataset
df.to_csv("Cleaned_Company_Employee_Dataset.csv", index=False)
print("\nSuccess! Cleaned dataset saved as 'Cleaned_Company_Employee_Dataset.csv'.")

--- Initial Dataset Shape ---
(157, 12)

--- Missing Values Before Cleaning ---
Employee_ID          0
Employee_Name        0
Department           5
Job_Title            0
Age                  4
Gender               5
Annual_Salary        5
Experience_Years     3
Joining_Date         0
City                 5
Performance_Score    3
Work_Mode            2
dtype: int64

Total duplicate rows detected: 7

--- Missing Values After Cleaning ---
Employee_ID          0
Employee_Name        0
Department           0
Job_Title            0
Age                  0
Gender               0
Annual_Salary        0
Experience_Years     0
Joining_Date         0
City                 0
Performance_Score    0
Work_Mode            0
dtype: int64

--- Final Cleaned Shape ---
(150, 12)

Success! Cleaned dataset saved as 'Cleaned_Company_Employee_Dataset.csv'.


Summary of Cleaning Steps Performed:

1.Initial Inspection: Loaded the raw dataset of 157 rows and 12 columns, identifying structural gaps, missing entries, and duplicate records.

2.Missing Value Treatment:

  Imputed missing numerical values (Age,
  Annual_Salary, Experience_Years, Performance_Score) using their respective medians to prevent skewness from potential outliers.

  Imputed missing categorical values (Department, Gender, City, Work_Mode) using their mode (most frequent category).

3.Duplicate Management: Identified and dropped 7 duplicate rows, reducing the dataset from 157 records down to a clean, unique set of 150 records.

4.Final Verification: Confirmed that missing value counts and duplicate counts dropped to zero across all columns prior to exporting the final CSV file.

